### DISCUSSION TITLE
Synchronization issue for large data set in neptune

### DISCUSSION THREAD

**Discussion by sanchit231** [aws/graph-explorer#76] [Q&A]
Hey, 
We have configured graph explorer on an EC2 instance. Connectivity is fine but we are having issues while synchronizing to Neptune from Graph explorer UI. We are pulling ~1.7 TB of data with 618 M nodes and 861 M Edges. Earlier we are getting timeouts so we have increased the query execution time to 30 mins. It ran but it takes a lot of time. Is there any way we can optimize this synchronization and expidite this process?

**Comment by triggan** [comment]
There is an optimization being developed to help with this, but it is dependent on changes to be made within the Neptune engine.  This should be addressed within Graph Explorer within the next couple of months, maybe sooner.  Really depends on the dependency within the Neptune engine being addressed.

We've also discussed (briefly) building some sort of method for users to manually define their graph "schema".  Further discussion is needed on whether this makes sense and how it could possibly be implemented.

**Comment by sanchit231** [comment]
What changes we have to make within Neptune? 
Maybe we can try that to check if it helps.

**Comment by triggan** [comment]
The capabilities within Neptune are not yet available.  These are engine changes that we (AWS) need to make within the service to enable this capability.
